include/swift/AST/Types.h

```diff
@@ -795,6 +795,9 @@ class alignas(1 << TypeAlignInBits) TypeBase
   /// Determine whether the type is an opened existential type with Error inside
   bool isOpenedExistentialWithError();
 
+  /// Determine whether the type is an Optional type.
```

Comment:
TypeBase already has an `isOptional`. It's defined by including "KnownStdlibTypes.def": https://github.com/swiftlang/swift/blob/737b5ec92433104ba975021f41e63da78723eff0/include/swift/AST/Types.h#L1039